Pages/IndieAuth/Auth.php

```diff
@@ -61,23 +61,62 @@ function post()
             $code         = $this->getInput('code');
             $client_id    = $this->getInput('client_id');
             $redirect_uri = $this->getInput('redirect_uri');
+            
+            $headers      = self::getallheaders();
 
             $verified = Auth::verifyCode($code, $client_id, $redirect_uri);
             if ($verified['valid']) {
-                $this->setResponse(200);
-                header('Content-Type: application/x-www-form-urlencoded');
-                echo http_build_query(array(
-                    'scope'        => $verified['scope'],
-                    'me'           => $verified['me'],
-                ));
+                $this->setResponse(200);  
+                if(!empty($headers['Accept'])) {
+                    switch($headers['Accept']) {
+                        case 'application/json':
```

Comment:
It may be worth performing more resilient `Accept` based negotiation (ie https://cweiske.de/tagebuch/php-http-negotiation.htm) as some clients may send an exact match

Reply:
Seems overkill to me. Following the spec only supporting application/json should be enough anyway. Just out of risk of breaking things (as Known was still on form-encoded) I left form-encoded as the default.

Reply:
Personally I'm not sure I agree. The HTTP Library I'd be using, [Apache HTTP Components for Java](https://hc.apache.org/) sends `application/json; charset=UTF-8` which therefore would not match.

There may also be cases where the `accept` header is in other formats, which may break perfectly valid clients

Reply:
Ah, didn't thought of the charset extension. Valid point than of course.

Reply:
Best solution would be to drop support for form-encoded responses completely I would think?